### PR TITLE
feat(application-system): example payment application and shared UI component for pending payments

### PR DESCRIPTION
**Pull request by arnorhs**
## What

- Fully working example project for how to add a payment step to an application.
  `/umsoknir/greida`
- Refactored the way we fetch payment redirect urls so they don't require an api module action to store specific data
- Added a shared component that can be reused in different applications

## Why

So people have some sort of reference thing to try out and perform payments and consult when they have issues/questions regarding payments

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by rafnarnason — APPROVED** [review]
lgtm ⚡

File: libs/application/template-api-modules/src/lib/modules/templates/example-payment-actions/examplePaymentActions.service.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  async submitApplication({
+    application,
+    auth,
+  }: TemplateApiModuleActionProps): Promise<{ success: boolean }> {
+    const paymentStatus = await this.sharedTemplateAPIService.getPaymentStatus(
+      auth.authorization,
+      application.id,
+    )
+
+    console.log({ paymentStatus })
```
[the remark this answers is not in the document]
Comment (arnorhs) [reply libs/application/template-api-modules/src/lib/modules/templates/example-payment-actions/examplePaymentActions.service.ts:52]: yes, ty

File: libs/application/template-api-modules/tsconfig.spec.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "**/*.spec.tsx",
     "**/*.spec.js",
     "**/*.spec.jsx",
-    "**/*.d.ts"
+    "**/*.d.ts",
+    "src/lib/modules/templates/example-payment-actions/examplePaymentActions.ts"
```
[the remark this answers is not in the document]
Comment (arnorhs) [reply libs/application/template-api-modules/tsconfig.spec.json]: honestly, no idea how that got added/changed..

[the remark this answers is not in the document]
Comment (saevarma) [reply libs/application/template-api-modules/tsconfig.spec.json]: I suspect Nx when creating new lib :P

**Review by saevarma — APPROVED** [review]

**Review by Thorrik58 — APPROVED** [review]
LGTM, but I have a general thought relating to example applications.

I can see that this example will help with implementing this flow in other applications and its more complex than other basic external services. 
But I do wonder what should be our standard for example applications, that is to say what warrants making an example application rather than just having clear documentation on how to implement and use certain features in the application system. 

I could see a problem with adding more and more applications with island.is growth, but I might be overthinking this. 
Maybe we could look into making one example applications where these examples could live or just changing the folder structure for them.

@eirikurn any thought on this? Am I overthinking this maybe 😅

**Comment by arnorhs** [comment]
> LGTM, but I have a general thought relating to example applications.
> 
> I can see that this example will help with implementing this flow in other applications and its more complex than other basic external services. But I do wonder what should be our standard for example applications, that is to say what warrants making an example application rather than just having clear documentation on how to implement and use certain features in the application system.
> 
> I could see a problem with adding more and more applications with island.is growth, but I might be overthinking this. Maybe we could look into making one example applications where these examples could live or just changing the folder structure for them.
> 
> @eirikurn any thought on this? Am I overthinking this maybe 😅

Yes, indeed. This is a telltale sign that the abstraction we implemented for including payments is too complicated. The whole thing was implemented under too much pressure as part of another application.

But as things stand there's a lot of tricky parts also in terms of the 3rd parties involved -  and I keep getting questions from people implementing payments (including people from sendiradid) where people are trying to look at the payment parts in the driving license application.

I did get approval from @vesteinnv for creating this application, and I do feel like at this point this should be merged. Next step is probably to planning the year for the payment feature in general. But even when doing that, I think this application will be helpful.

**Comment by Thorrik58** [comment]
@arnorhs 
Yep I have been there and I have nothing against this application example and I see you have done all the due diligence therefore the approve 😄 

This is more a general thought regarding the future cases that might come up.

**Pull request merged**